android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java

```diff
@@ -1093,7 +1093,15 @@ private void dispatchCallback(String name, KrollDict data) {
 
 		data.put("source", activityProxy);
 
-		activityProxy.callPropertyAsync(name, new Object[] { data });
+		// TIMOB-19903
+		if (TiApplication.getInstance().runOnMainThread()) {
+			// We must call this synchornously to ensure it happens before we release the Activity reference on the V8/Native side!
+			activityProxy.callPropertySync(name, new Object[] { data });
+		} else {
+			// This hopefully finishes before we release the reference on the native side?! I have seen it crash because it didn't before though...
+			// Not sure it's safe to keep this behavior...
+			activityProxy.callPropertyAsync(name, new Object[] { data });
```

Comment:
Ah, I'm not up to date with the recent hyperloop changes. If the callbacks are async then there's no guarantee that it will be finished before release call. This change should be fine.